src/object/srv_ec_aggregate.c

```diff
@@ -1085,6 +1085,7 @@ agg_update_parity(struct ec_agg_entry *entry, uint8_t *bit_map,
 			goto out;
 		while (!isset(bit_map, j))
 			j++;
+		D_ASSERTF(j < k, "bad cell_idx %d, number of data shards %d\n", j, k);
```

Comment:
Strictly speaking, if really out of boundary, above check cannot work. We may have to check `j` before `++`, means:
```
while (j < k && !isset(bit_map, j))
    j++;
D_ASSERTF(j < k, "bad cell_idx %d, number of data shards %d\n", j, k);
```
Anyway, we can adjust it when have other chance to refresh the patch.

Reply:
right, can change it when need to refresh it later, or by other PR, thanks